doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -389,15 +386,13 @@ the propositional implication and function types.
 Applications
 ------------
 
-The expression :token:`term`\ :math:`_0` :token:`term`\ :math:`_1` denotes the
-application of :token:`term`\ :math:`_0` to :token:`term`\ :math:`_1`.
+The expression :n:`@term__0 @term__1` denotes the
+application of :n:`@term__0` to :n:`term__1`.
```

Comment:
FWIW, this description is underwhelming.  Presumably term__0 has to evaluate to a function and can't be, say `1`.

Also might be better to use a name other than `term__0` since it is fundamentally different from the other term__i.  How about something like ![image](https://user-images.githubusercontent.com/1253341/71538749-5d876e80-28e5-11ea-86da-a86a0f9f3013.png)?

I hope we're consistent in using 0- or 1-origin indexing for subscripts throughout the doc.  I think I'd lean toward using 1-origin.

Reply:
> The fact that it must be a function is given by the typing. And no it doesn't have to "evaluate" to a function, as it could be an axiom.

I may not be using the common Coq nomenclature, but it would be good to say that term__0 has to have a function type.  And IMHO calling it, say, term__fun makes the distinction more clearly than term__0 (names are usually more understandable than magic numbers).  My "FWIW" above was meant to imply it's your choice whether to this in this PR.

Reply:
Also, the second line is missing an @ sign: it should have ```:n:`@term__1` ```

![image](https://user-images.githubusercontent.com/1253341/71549072-3cb62c00-296c-11ea-8666-e45f7d29238f.png)